addons/hr/models/hr_employee.py

```diff
@@ -505,10 +505,16 @@ def _post_author(self):
     def _get_unusual_days(self, date_from, date_to=None):
         # Checking the calendar directly allows to not grey out the leaves taken
         # by the employee or fallback to the company calendar
-        return (self.resource_calendar_id or self.env.company.resource_calendar_id)._get_unusual_days(
-            datetime.combine(fields.Date.from_string(date_from), time.min).replace(tzinfo=UTC),
-            datetime.combine(fields.Date.from_string(date_to), time.max).replace(tzinfo=UTC)
-        )
+        if self:
+            domain = [('company_id', '=', self.company_id.id)]
+        else:
+            domain = [('company_id', '=', self.env.company.id)]
```

Comment:
As this method doesn't seem to be made to give data for more than one employee, let's add an `ensure_one()`.
Additionally, we also need to fetch data if company_id is `False` as that field is not required:
```suggestion
        self.ensure_one()
        if self:
            domain_company = self.company_id.id
        else:
            domain_company = self.env.company.id
        domain =  ['|', ('company_id', '=', False), ('company_id', '=', domain_company)]
```